62/serfer2/pycon_es.py

```diff
@@ -0,0 +1,155 @@
+PYCON_ES_SOURCES = (
+    {
+        'year': '2018',
+        'url': 'https://2018.es.pycon.org/#schedule',
+        'xpath': '//div[@class="schedule__talk--speaker"]/div/p/text()'
+    }, {
+        'year': '2017',
+        'url': 'https://2017.es.pycon.org/es/schedule/',
+        'xpath': '//div[@class="slot-inner"]/p/strong/text()'
+    }, {
+        'year': '2016',
+        'url': 'http://2016.es.pycon.org/es/schedule/',
+        'xpath': '//div[@class="slot-inner"]/p/strong/text()'
+    }, {
+        'year': '2015',
+        'url': 'http://2015.es.pycon.org/es/schedule/',
+        'xpath': '//div[@class="slot-inner"]/p/strong/text()'
+    }, {
+        'year': '2014',
+        'url': 'http://2014.es.pycon.org/talks',
+        'xpath': '//div[@id="onlytext"]/h1[@class="text-center"]/text()'
+    },
+    {
+        'year': '2013',
+        'url': 'http://2013.es.pycon.org/#agenda',
+        'xpath': '//ul[@class="schedule-list"]/li/div/a[@target="_blank"]/text()'
+    }
+)
+KNOWN_GENDERS = ['male', 'female', 'unknown']
+
+
+def get_gender(name):
+    """
+    Gets name gender from https://gender-api.com
+    """
+    name_data = json.loads(requests.get(f'https://gender-api.com/get?name={name}&key={GENDER_API_KEY}').content)
+    return name_data.get('gender', '')
+
+
+def load_names_cache():
+    """
+    Read local files with names by gender
+    """
+    names = dict()
+    with open(NAMES_CACHE_FNAME, 'r') as fi:
+        lines = fi.readlines()
+        for line in lines:
+            line = line.strip()
+            if not line:
+                continue
+            name, gender = line.split(';')
+            names[name] = gender
+    return names
+
+
+def save_names_cache(names):
+    """
+    Saves names and gendes in CSV
+    """
+    with open(NAMES_CACHE_FNAME, 'w') as fo:
+        for name, gender in names.items():
+            fo.write(f'{name};{gender}\n')
+
+
+def save_results(results):
+    """
+    saves resulting data to CSV
+    """
+    with open(RESULTS_FNAME, 'w') as fo:
+        fo.write('year,total,male,male %,female,female %\n')
+        for year, res in results.items():
+            female_pcnt = 0
+            male_qty = len(res.get('male', []))
+            female_qty = len(res.get('female', []))
+            qty = female_qty + male_qty
+            if qty > 0:
+                female_pcnt = float(female_qty * 100.0 / qty)
+                male_pcnt = float(male_qty * 100.0 / qty)
+            line = '%s,%d,%d,%.02f,%d,%.02f\n' %\
+                (year, qty, male_qty, male_pcnt, female_qty, female_pcnt)
+            fo.write(line)
+
+
+def viz_data():
+    """
+    Result data visualization
+    """
+    data_frame = pandas.read_csv(RESULTS_FNAME).sort_values('year')
+    data_frame.plot(
+        x='year',
+        y=['male %', 'female %'],
+        legend=True,
+        title='speaker\'s gender evolution (%)',
+        xticks=data_frame['year'].tolist(),
+        kind='bar'
+    )
+    plt.show()
```

Comment:
instead of calling `plt.show` you could consider returning the figure object which allows the caller to further change the graphical output